src/OrchardCore/OrchardCore.Email.Abstractions/MailMessage.cs

```diff
@@ -46,13 +46,7 @@ public class MailMessage
     /// <summary>
     /// Gets or sets the message content aka body.
     /// </summary>
-    /// <remarks>This property is work in conjunction with <see cref="IsHtmlBody"/> to determine the body type..</remarks>
-    public string Body { get; set; }
```

Comment:
We should also make these obsolete instead of removing immediately. The properties can be forwarded to the new body, which has to have a different name, e.g. `MailBody`.

Reply:
The problem is needs a logic because we remove two properties, moreover we need to introduce a new name such as `MailBody`. I don't think it's an issue while the next release is major

Reply:
Then at least document that breaking change correctly. 

To avoid this breaking change, we could add a second body property to the mail message directly, e.g. `AlternateBody` and keep the current `Body` and `IsHtmlBody` as is. This new `AlternateBody` property would contain the opposite format as set by `IsHtmlBody`. Just an idea, as we had some pushbacks against to many breaking changes lately.

Reply:
`AlternateBody` is a good option too, but `IsHtmlBody` is not suited because we might send both Text & HTML. We could use nullable for `IsHtmlBody`

Reply:
For simplicity, I changed the docs to clarify there's a breaking change in the return type

Reply:
> `AlternateBody` is a good option too, but `IsHtmlBody` is not suited because we might send both Text & HTML. We could use nullable for `IsHtmlBody`.

The `AlternateBody` would contain plain text, if `IsHtmlBody` is true, otherwise it would contain HTML. If it's null, only one body type would be send. No changes to `IsHtmlBody` would be required.

I acknowledge that my idea is not the greatest API shape, but it avoids a breaking change in the mail message.

Reply:
To further clarify what I mean: The `AlternateBody` property would contain a string and would be used instead of the new type `MailMessageBody`.

Reply:
I understand what you mean, but we need to deal with two properties instead of one, I just like the `MailKit` body, but I'm open to any suggestion, @sebastienros your thought to finalize this PR

Reply:
I know we need two body properties, but there is no need to create a new class for them. Why not just obsolete the `Body` and `IsHtmlBody` properties and create the two new properties directly on the mail message, e.g. as `TextBody` and `HtmlBody`. This can be made without a breaking change for now.

Reply:
It could be, but we should have a way to decide which is the mail format should be used, HTML, text or both

Reply:
Couldn't that be as simple as testing the properties for null or emptiness?